src/lib/axios-logger.ts

```diff
@@ -0,0 +1,44 @@
+import axios, { AxiosError } from "axios";
+
+import MailtrapError from "./MailtrapError";
+
+import { AxiosErrorObject } from "../types/axios";
+
+/**
+ * Axios error names guard.
+ */
+const hasErrorProperty = <T extends keyof AxiosErrorObject>(
+  obj: unknown,
+  propertyName: T
+): obj is AxiosErrorObject => {
+  return (obj as AxiosErrorObject)?.[propertyName] !== undefined;
+};
+
+/**
+ * Error handler for axios response.
+ */
+export default function handleSendingError(error: AxiosError | unknown) {
+  if (axios.isAxiosError(error)) {
+    const serverErrorsObject =
+      error.response?.data &&
+      typeof error.response.data === "object" &&
```

Comment:
👍 

Great that you're handling it. It looks that our API docs don't mention it, but we have 422 error responses which usually return an object.